DevProxy.Abstractions/Plugins/BasePlugin.cs

```diff
@@ -99,6 +99,15 @@ public TConfiguration Configuration
     }
     public IConfigurationSection ConfigurationSection { get; } = pluginConfigurationSection;
 
+    protected virtual IEnumerable<string>? GetConfigurationValue(string key, IEnumerable<string>? list, IEnumerable<string>? defaultList = default)
+    {
+        if (key is null) { return null; }

```

Comment:
Key shouldn't be `null`. I suggest we change this to throw an ArgumentNullException.